src/bin/build_octree.rs

```diff
@@ -382,4 +402,32 @@ fn main() {
         // their parents.
         nodes_to_subsample.extend(subsample_nodes.into_iter());
     }
+
+    // Add all node meta data to meta.pb
+    // TODO(tschiwietz): How can we make sure not to miss any messages?
+    while let Ok((id, node_meta)) = nodes_receiver.try_recv() {
```

Comment:
This should work:

drop(nodes_sender); 
for node_meta in nodes_receiver {}

The stream terminates as soon as no sender is live anymore. All the clones die with their scope, but the original is still live. You can drop it manually to avoid a deadlock.